docs/docs-beta/docs/guides/build/components/setting-up-a-deployment.md

```diff
@@ -0,0 +1,177 @@
+---
+title: "Setting up a deployment"
+sidebar_position: 200
+unlisted: true
+displayed_sidebar: 'components'
+---
+
+Many users will want to manage multiple code locations within a single coherent
+directory structure. `dg` facilitates this with the concept of a _deployment_
+directory.
+
+A deployment directory contains a root `pyproject.toml` containing
+deployment-level settings and a `code-locations` directory containing one or
+more code locations. A deployment does _not_ define a Python environment by
+default-- instead, Python environments are defined per code location.
+
+To scaffold a new deployment, run:
+
+```bash
+$ dg deployment scaffold my-deployment
+
+Creating a Dagster deployment at my-deployment.
+Scaffolded files for Dagster project in my-deployment.
+```
+
+This will create a new directory `my-deployment`. Let's look at the structure:
+
+```bash
+$ cd my-deployment && tree
+
+.
+├── code_locations
+└── pyproject.toml
+```
+
+Importantly, the `pyproject.toml` file contains an `is_deployment` setting
+marking this directory as a deployment:
+
+```toml
+### pyproject.toml
+
+[tool.dg]
+is_deployment = true
+```
+
+To add a code location to the deployment, run:
+
+```bash
+$ dg code-location scaffold code-location-1
+
+Creating a Dagster code location at .../my-deployment/code_locations/code-location-1.
+Scaffolded files for Dagster project in .../my-deployment/code_locations/code-location-1.
+...
+```
+
+This will create a new directory `code-location-1` within the `code_locations`.
+It will also setup a new uv-managed Python environment for the code location. Let's have a look:
+
+```bash
+$ tree
+
+├── code_locations
+│   └── code-location-1
+│       ├── code_location_1
```

Comment:
isn't there a `.venv` file in here too?

Reply:
Yes, but hidden from output by default because it's a dotfile. Should we show it?

Reply:
might want to note it in a comment or have tree spit out hidden files. seeing the `.venv` I think makes `uv` "click" with a lot of users.

Reply:
I'll add in a comment because the venv contains tons of files and it would be confusing not to show them as well if we show `.venv`